network_simulator/main_event_loop.py

```diff
@@ -0,0 +1,71 @@
+"""
+Module for the MainEventLoop class.
+"""
+
+from Queue import PriorityQueue
+import logging
+
+from .statistics import *
+
+logger = logging.getLogger(__name__)
+
+
+class MainEventLoop(object):
+    """
+    An event loop that holds all of the events that are going to occur,
+    and also keeps track of Statistics for the network.
+    """
+
+    def __init__(self):
+        """
+        :ivar float global_clock_sec: global clock that advances on
+        completion of Events.
+        :ivar PriorityQueue events: a priority queue of (start_time_sec,
+        Event) tuples, used to easily retrieve the (earliest) next Event.
+        :ivar Statistics statistics: statistics for this network.
+        :return:
```

Comment:
removed. they're put in by default in PyCharm when I start adding documentation.